### PR TITLE
feat: added get job by job parameters support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -152,6 +152,23 @@ paths:
                 $ref: '#/components/schemas/errorMessage'
       tags:
         - jobs
+  /jobs/parameters:
+    get:
```
Comment (asafMasa) [inline openapi3.yaml:156]: shouldn't it be POST?

File: src/DAL/migration/v3.0.0.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CREATE INDEX "jobParametersIndex"
+  ON "JobManager"."Job" USING btree
+  (parameters ASC NULLS LAST);
```
Comment (asafMasa) [inline src/DAL/migration/v3.0.0.sql]: end of file

File: src/common/utils.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export const paramsQueryBuilder = (params: Record<string, unknown>): string => {
+  const queryStringArray: string[] = [];
```
Comment (asafMasa) [inline src/common/utils.ts:3]: add // TODO: need to support nested paremeters, currently only 1 level is supported

File: src/jobs/controllers/jobController.ts [hunk 4]
```diff
@@ -48,6 +50,15 @@ export class JobController {
     }
   };
 
+  public getJobByJobsParameters: GetJobsByJobsParametersHandler = async (req, res, next) => {
```
Comment (asafMasa) [inline src/jobs/controllers/jobController.ts:53]: add Unit & Integration test

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job!

File: openapi3.yaml (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:156]: been discussed

File: src/DAL/migration/v3.0.0.sql (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/DAL/migration/v3.0.0.sql]: fixed

File: src/common/utils.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/utils.ts:3]: added

File: src/jobs/controllers/jobController.ts (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/jobs/controllers/jobController.ts:53]: added

**Review by asafMasa — APPROVED** [review]
👍

**Review by netanelC — APPROVED** [review]

**Pull request merged**